src/sage/modular/arithgroup/congroup_gamma1.py

```diff
@@ -650,3 +657,246 @@ def dimension_new_cusp_forms(self, k=2, eps=None, p=0, algorithm="CohenOesterle"
         eps_p = eps.restrict(N//p)
         old = Gamma1_constructor(N//p).dimension_cusp_forms(k, eps_p, algorithm)
         return self.dimension_cusp_forms(k, eps, algorithm) - 2*old
+
+
+
+def _dimension_new_cusp_forms_Ross(N,k,chi):
+    r"""An internal function to implement the explicit dimension formula
+        given in Theorem 1.4 of https://arxiv.org/abs/2407.08881. For tests, 
+        see test_dim_formula_Ross() in sage/modular/arithgroup/tests.py  
+    """
```

Comment:
```suggestion
    r"""
    Compute the dimension formula given in Theorem 1.4 of :arxiv:`2407.08881`.
    
    For more tests, see :meth:`sage.modular.arithgroup.tests.Test.test_dim_formula_Ross()`.
    """
```
However this function does need at least one doctest of its own.